### PR TITLE
Fix thread safety of bf, bmv2 and np4 ChassisManager

### PR DESCRIPTION
**Pull request by pudelkoM**
Uncovered by #306

### CODE REVIEW

File: stratum/hal/lib/bmv2/bmv2_chassis_manager.cc [hunk 1]
```diff
@@ -521,9 +523,9 @@ ::util::Status Bmv2ChassisManager::Shutdown() {
   // It is fine to release the chassis lock here (it is actually needed to call
   // UnregisterEventWriters or there would be a deadlock). Because initialized_
   // is set to true, RegisterEventWriters cannot be called.
-  APPEND_STATUS_IF_ERROR(status, UnregisterEventWriters());
```
Comment (bocon13) [inline stratum/hal/lib/bmv2/bmv2_chassis_manager.cc:524]: Should we look at the comment above?

Also, if we do decide to keep the lock to unregister the event writer, then we should probably just merge this with the block above.

File: stratum/hal/lib/np4intel/np4_chassis_manager.cc [hunk 2]
```diff
@@ -408,9 +407,9 @@ ::util::Status NP4ChassisManager::Shutdown() {
   // It is fine to release the chassis lock here (it is actually needed to call
   // UnregisterEventWriters or there would be a deadlock). Because initialized_
   // is set to true, RegisterEventWriters cannot be called.
-  APPEND_STATUS_IF_ERROR(status, UnregisterEventWriters());
```
Comment (bocon13) [inline stratum/hal/lib/np4intel/np4_chassis_manager.cc:411]: Same comment as bmv2

File: stratum/hal/lib/bmv2/bmv2_chassis_manager.cc (hunk 1 above)
Comment (pudelkoM) [reply stratum/hal/lib/bmv2/bmv2_chassis_manager.cc:524]: mhh, good catch.

I'll have to the verify the lock usage again.

**Comment by pudelkoM** [comment]
PR #306 added exclusion for these affected classes. Currently we do not observe any synchronization issues and postpone this PR to a later date.

**Pull request closed without merging**